### DISCUSSION TITLE
My robot's map keeps rotating by ~45 degrees rendering saved maps unusable

### DISCUSSION THREAD

**Discussion by skruehoetl** [Hypfer/Valetudo#2007] [Q&A / Support]
As stated in the title, I've got persistent-map-issues with my Commodore CVR 200:

- I enabled persistent maps in the settings
- The robot does a cleanup-all and scans the room
- In the newly created map, I define a few segments and walls
- The robot can clean up the segments as expected
but then
- after a few days, the robot starts "thinking" it's map is rotated by (typically) 45 degrees so the segments don't make sense and it keeps bumping into walls and tables.

I've reset the map a few times now and it would be great to have a permanent fix. Any advice?

**Comment by Hypfer** [comment]
This isn't really related to Valetudo as Valetudo is just a cloud replacement that happens to run on the robot and thus has no impact on any mapping or routing or any other business logic.

Regardless, this is a somewhat known issue with vacuum robots in general.
Some robots/firmwares cope with this better than others and unfortunately the CVR 200 is a budget model with a rather old firmware, meaning that it will be especially prone to this.

It is caused by the odometry data being incorrect but correct enough that it isn't flagged as incorrect.
This means that a wheel slips or the robot gets stuck on something but doesn't notice that it happened and instead changes the map.

Look for loose rugs, furniture it might get stuck on or very slippery tiles.
Fixing those hazards will also stop the map from deteriorating.

**Reply by skruehoetl** [reply]
Okay, not the answer I was hoping for, but a good answer nevertheless. Strangely, it doesn't seem to happen anymore since my last "delete map data" operation. The only thing, I did different this time was that I refrained from setting up virtual restrictions (walls, keep-out zones) but that might just be a coincidence. (I am aware that it's just a budget toy I'm talking about but with the nice Valetudo-Home-Assistant integration it is already far more useful than 1 thought!)

Thanks for helping out!
